### PR TITLE
fix Whisper tests on GPU

### PR DESCRIPTION
**Pull request by hollance**
# What does this PR do?

The daily CI showed that Whisper has new test failures, related to the recent merge of the prompting feature. This PR fixes those test failures.

The tests ran OK on CPU but failed on GPU because the input data wasn't moved to the GPU.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by hollance** [comment]
Also skipping a few tests in `WhisperModelTest` that were previously skipped in `WhisperEncoderModelTest`, see https://github.com/huggingface/transformers/pull/22060

**Review by sgugger — COMMENTED** [review]
Thanks for the fix, just one question on the model parallelism tests.

File: tests/models/whisper/test_modeling_whisper.py [hunk 1]
```diff
@@ -393,6 +393,18 @@ def test_inputs_embeds(self):
             with torch.no_grad():
                 model(**inputs)[0]
 
+    @unittest.skip(reason="Some undefined behavior encountered with tiny versions of this model. Skip for now.")
```
[the remark this answers is not in the document]
Comment (hollance) [reply tests/models/whisper/test_modeling_whisper.py:396]: Looking through the daily-ci history I don't actually see any WhisperModelTest failures on `test_model_parallelism`, but `test_cpu_offload` and/or `test_disk_offload` do seem to fail occasionally. These tests were disabled in WhisperEncoderModelTest, so I also disabled them here. They seem to be unrelated to the prompting PR.

**Review by sanchit-gandhi — APPROVED** [review]
LGTM - thanks for the torch device fixes @hollance!

File: tests/models/whisper/test_modeling_whisper.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sgugger) [reply tests/models/whisper/test_modeling_whisper.py:396]: Let's leave them as is then.

[the remark this answers is not in the document]
Comment (hollance) [reply tests/models/whisper/test_modeling_whisper.py]: OK, removed those tests. Feel free to merge at your leisure.

**Comment by sgugger** [comment]
Thanks again!

**Pull request merged**